source/sync/error-handling/errors.txt

```diff
@@ -148,19 +148,36 @@ The following errors may occur when your {+app+} uses :ref:`Flexible Sync
        The client attempts to reconnect until this process is complete. Then,
        this error resolves and Sync begins working normally.
     
-   * - ErrorWriteNotAllowed
```

Comment:
Sorry, I should have been more explicit about this in the ticket, but this error is still being used under the following conditions:

1. A client performs an illegal change that _would_ trigger a compensating write, but their SDK version does not support the non-fatal nature of compensating writes.
    - the behavior would match the current description
2. A client performs a change on an asymmetric table that violates their write permissions, and their SDK version includes [this change](https://github.com/realm/realm-core/pull/5690)
    - the behavior would match the current description, except it will be non-fatal and not trigger a client reset (@benredmond has an open PR for this)

I'm not sure if there's precedence for documenting behavior of old SDK versions. If not then bullet 1 above can be ignored

Reply:
ah just noticed the callout about (1) in the compensating write error description

Reply:
Concerned about mentioning asymmetric tables as this would be the first mention and would probably be more confusing. Is there a specific scenario I can describe where this would happen?

Reply:
Hm, there's no scenario that doesn't involve asymmetric tables. It's only if you try to create an object on an asymmetric table that you don't have write permissions for.

Reply:
I see. So, what's an asymmetric table?

Reply:
Ah! Of course. 💡